mllib/src/main/scala/org/apache/spark/ml/r/RWrapperUtils.scala

```diff
@@ -35,13 +35,37 @@ object RWrapperUtils extends Logging {
    */
   def checkDataColumns(rFormula: RFormula, data: Dataset[_]): Unit = {
     if (data.schema.fieldNames.contains(rFormula.getLabelCol)) {
-      logWarning("data containing 'label' column, so change its name to avoid conflict")
-      rFormula.setLabelCol(rFormula.getLabelCol + "_output")
+      val newLabelName = convertToUniqueName(rFormula.getLabelCol, data.schema.fieldNames)
+      logWarning(
+        s"data containing ${rFormula.getLabelCol} column, changing its name to $newLabelName")
```

Comment:
this sounds a bit like we are renaming the existing `label` column?
perhaps just change to `s"data containing ${rFormula.getLabelCol} column, using new name to $newLabelName instead"`?